ste/mgr-JobMgr.go

```diff
@@ -875,8 +937,13 @@ func (jm *jobMgr) transferProcessor(workerID int) {
 	for {
 		// No scaleback check here, because this routine runs only in a small number of goroutines, so no need to kill them off
 		select {
+		case <-jm.Context().Done():
+			fmt.Println("transferProcessor done called")
+			return
```

Comment:
Again, could it so happen that we still have some transfers in the channel.
In general do you think it'll be safe to de deferred close for everything, i.e. instead of writing to done channel right away, right after 60 seconds so we are safe from such issues and it's ok if the resources are freed after a minute.

Reply:
As discussed, We will be doing cleanup after moving entry to completed or failed folder. Which is like 300sec after job completed.

Reply:
@nitin-deamon, I think it's still better to use the deferred close in the code since this is a generic feature and the correctness should not depend on a specific caller taking care of calling it after a delay. Once we have the jobmgr cleanup support, tomorrow we can have some other caller call it right after the job completes.

Reply:
It need to be done as part of ReportHandler, as only that function knows when job is done. 
We can call cleanup function from reportHandler, and it will take care of it.

Reply:
Sorry I don't understand the ReportHandler comment.
All I'm asking is instead of calling cleanup "right now" defer it for sometime in JobMgrCleanUp().
So JobMgrCleanUp() calls ja.DeleteJob() promptly so that the job is  no more externally reachable, but it can defer rest of the cleanups by calling it from a go routine with a 60 secs sleep in the beginning.
That way we are sure that after a job is removed from jobmgr and hence no further activity can take place in the job, we cleanup other things after a quiesce period.
Note that all of this is needed because there are comments in the code which indicate that there could be data coming on the channels even after close/cleanup.
This deferred cleanup is a safe way to handle it w/o worrying about the actual fix, which we should do when we can.

Reply:
Rather than waiting for a fixed amount of time, perhaps it's better to actively drain all the channels, and make sure nothing stays after the cleanup.

@nakulkar-msft thoughts on this?

Reply:
@zezha-msft, properly draining the channel(s) is good and needed, and that would be the right way to kind of fix the TODO.
Since the prev TODO gives an impression that someone adds new entries later and we didn't know how (or didn't have enough debug info) I suggested the wait because the reader can drain the channel but he cannot be sure about whether a writer can add something afterwards, unless we ensure that there are no writers writing afterwards.